### PR TITLE
Split `ProtostarFixture` file

### PR DESCRIPTION
**Pull request by kasperski95**
Make the file smaller.

### CODE REVIEW

**Comment by MaksymilianDemitraszek** [comment]
I would restrain from such refactors ATM it may interfere with stuff @Arcticae is working on.

**Comment by kasperski95** [comment]
This change should be unrelated to Cairo test runner.

File: tests/integration/_conftest/spying_gateway_facade_factory.py
Comment (karol-bisztyga) [inline tests/integration/_conftest/spying_gateway_facade_factory.py:13]: Just out of curiosity: where'd you come up with such a name? Why "spying"?

**Review by karol-bisztyga — COMMENTED** [review]
I think it looks ok, but I'd wait for one more review

File: tests/integration/_conftest/spying_gateway_facade_factory.py
Comment (kasperski95) [reply tests/integration/_conftest/spying_gateway_facade_factory.py:13]: Because it uses `GatewayClientTxInterceptor` instead of `GatewayClient`. (Maybe `SpyingGatewayClient` would be a better name).  That client puts every transaction into `TransactionRegistry` which can be later used by tests to analyze what is actually sent.

I renamed it from `TestFriendyGatewayFacadeFactory`, because `spy` seems a better word than `test friendly`.

> Spies are stubs that also record some information based on how they were called.
https://martinfowler.com/bliki/TestDouble.html

**Comment by kasperski95** [comment]
@Arcticae @MaksymilianDemitraszek ping

File: tests/integration/_conftest/transaction_registry.py
Comment (Arcticae) [inline tests/integration/_conftest/transaction_registry.py:8]: Why bother with 2 separate classes doing basically the same thing? (capturing the txns)

File: tests/integration/_conftest/gateway_client_tx_interceptor.py
Comment (Arcticae) [inline tests/integration/_conftest/gateway_client_tx_interceptor.py:12]: why make the tx registry optional?

**Review by Arcticae — CHANGES_REQUESTED** [review]
I don't see why we should keep 2 classes which one of them only keeps the txns sorted by the type. I'd rather have a getter method which returns transactions of a certain type.

File: tests/integration/_conftest/transaction_registry.py
Comment (kasperski95) [reply tests/integration/_conftest/transaction_registry.py:8]: `GatewayClient` is instantiated multiple times (each network command creates a `GatewayClient`). `TransactionRegistry` stores transactions from all gateway clients instantiated within a single test. 

These classes already exist in the code base. This PR splits the `protostar_fixture.py` file.

File: tests/integration/_conftest/gateway_client_tx_interceptor.py
Comment (kasperski95) [reply tests/integration/_conftest/gateway_client_tx_interceptor.py]: [remove optional from transaction registry](https://github.com/software-mansion/protostar/pull/1298/commits/311602d4daa3d52eb9d0d0477fe65a6ddb6930e9)

**Review by Arcticae — APPROVED** [review]

**Pull request merged**
